File: Editorial/0106_Construct_Binary_Tree_from_Inorder_and_Postorder_Traversal/Go/1_Recursion.go
func buildTree(inorder []int, postorder []int) *TreeNode {
    idx_map := map[int]int{}
    post_idx := len(postorder) - 1
    var helper func(in_left int, in_right int) *TreeNode
    helper = func(in_left int, in_right int) *TreeNode {
        if in_left > in_right {
            return nil
        }
        root_val := postorder[post_idx]
        root := &TreeNode{Val: root_val}
        index := idx_map[root_val]
        post_idx--
        root.Right = helper(index+1, in_right)
        root.Left = helper(in_left, index-1)
        return root
    }
    for i := 0; i < len(inorder); i++ {
        idx_map[inorder[i]] = i
    }
    return helper(0, len(inorder)-1)
}

File: Editorial/106_Construct_Binary_Tree_from_Inorder_and_Postorder_Traversal/JavaScript/1_Recursion.js
var buildTree = function (inorder, postorder) {
    var idx_map = {};
    var post_idx = postorder.length - 1;
    var helper = function (in_left, in_right) {
        if (in_left > in_right) return null;
        var root_val = postorder[post_idx];
        var root = new TreeNode(root_val);
        var index = idx_map[root_val];
        post_idx--;
        root.right = helper(index + 1, in_right);
        root.left = helper(in_left, index - 1);
        return root;
    };
    for (var i = 0; i < inorder.length; i++) idx_map[inorder[i]] = i;
    return helper(0, inorder.length - 1);
};
